src/topology/algebra/mul_action2.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2021 Alex Kontorovich, Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Kontorovich, Heather Macbeth
+-/
+import topology.homeomorph
+import group_theory.group_action.basic
+/-!
+# Continuous monoid action
+
+In this file we define class `has_continuous_smul₂`. We say `has_continuous_smul₂ Γ T` if `Γ` acts
+on `T` and for each `γ`, the map `x ↦ γ • x` is continuous.
+
+## Main definitions
+
+* `has_continuous_smul₂ Γ T` : typeclass saying that the map `x ↦ γ • x` is continuous on `T`;
+* `properly_discontinuous_smul`: says that the scalar multiplication `(•) : Γ → T → T`
+is properly discontinuous, that is, for any pair of compact sets `K, L` in `T`, only finitely many
+`γ:Γ` move `K` to have nontrivial intersection with `L`.
+* `homeomorph.smul`: scalar multiplication by an element of a group `Γ` acting on `T`
+  is a homeomorphism of `T`.
+
+## Main results
+
+* `is_open_map_quotient_mk_mul` : The quotient map by a group action is open.
+* `is_t2_of_properly_discontinuous_smul_of_t2` : The quotient by a discontinuous group action of
+a locally compact t2 space is t2.
+
+## Tags
+
+Hausdorff, discrete group, properly discontinuous, quotient space
+
+-/
+
+local attribute [instance] mul_action.orbit_rel
+
+/-- Class `has_continuous_smul₂ Γ T` says that the scalar multiplication `(•) : Γ → T → T`
+is continuous in the second argument. We use the same class for all kinds of multiplicative
+actions, including (semi)modules and algebras.
+-/
+class has_continuous_smul₂ (Γ : Type*) (T : Type*) [topological_space T] [has_scalar Γ T]
```

Comment:
also, I'm not sure about the ₂ naming, but can't come up with a better idea; I'd suggest `has_continuous_lsmul`, but that's usually for `smul` as a linear map...

Reply:
This name is not consistent with https://leanprover-community.github.io/mathlib_docs/measure_theory/group/arithmetic.html#has_measurable_smul

Reply:
Oh, I didn't know there was precedent for this ₂ naming. Nevermind!

Reply:
I meant that the naming in this PR is not consistent with `has_measurable_smul` (the meaning of `₂` is the opposite).

Reply:
@urkud That's true, although unfortunately the existing meanings of both `has_continuous_smul` and `has_measurable_smul` have been around for a while.  Do you think we should change it for consistency?  If so, do you have an opinion on which one should be `₂`?